### ISSUE TITLE
Resting causes unresponsiveness

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#87]
```
What steps will reproduce the problem?
1. Go to a bed and rest for free
2. Game comes unresponsive and the rest dialog, with "Rest" button activated
3. Android shows "force close" and choosing it, returns game to app start 
(savegame not load)

What is the expected output? What do you see instead?
Resting without a problem as before.

What version of the product are you using? On what operating system?
v0.6.6 on HTC Wildfire, Android 1.37.

Please provide any additional information below.
I am level 7 with 4/30 HP. No idea why I am unable to rest.
Is there a way I can provide more detailed information ?
```

Original issue reported on code.google.com by `kims...@gmail.com` on 15 Dec 2010 at 4:06

**Comment by GoogleCodeExporter** [comment]
```
Upgraded to v0.6.7a and problem still exist. Any rest, free or paid, as well as 
being attacked triggers this.
Once I restart game am I able to do a number of things. Move around, enter/exit 
areas, talk to NPCs, begin fights and complete them, flee from such fights, 
pickup loot from fights, change prefs, change gear, buy/sell items.

"Loading resources.." stays on screen much longer now than in v0.6.6.

Loaded quick savegame and was in battle. Had 3 new issues:
* The screen is off my view, can just see my HP bar. Opening "Info on monster" 
corrected it.
* Once fixed, I changed to landscape and the monster HP showed as 1/5th full 
but text said [Max HP]/[Max HP] - damaging the monster corrected this graphic 
issue.
* When I am at 0 HP, dead, it also becomes unresponsive :(

I was able to do several hit-and-flee attacks. Wasnt this fixed ?

I have a savegame on SD card, will this be lost if I uninstall the game ?
```

Original comment by `kims...@gmail.com` on 15 Dec 2010 at 11:33

**Comment by GoogleCodeExporter** [comment]
```
Lots of separate issues in this one. Thanks for reporting it, but try to keep 
each thread about one issue.

About the resting that is causing FC:s for you. I would be very interested in 
getting your savegame to examine if there's any problem with the spawning of 
monsters on your savegame. Could you please attach the savegame to this issue? 
If you run v0.6.7b1, you can save the game to four slots, where each slot is 
saved as a separate file on your sd card.

Regarding "Loading resources.. stays on screen": yes, there are now several new 
maps that has to be loaded, which takes considerable time. You should see it on 
the emulator :) We might change this in the future to only load maps when you 
actually travel to them. But for the moment, all maps are loaded when you start 
the game, so you can change maps instantaneously.

The error with HP bars when changing to landscape mode is reported in issue 4. 
Thanks for noticing it again. We haven't gotten around to fixing it yet since 
it's purely cosmetical.

Unresponsive when 0HP=dead is probably the same problem as when resting, since 
it's almost the same code that is being run.

Savegames on SD card will not be lost when uninstalling the game.
```

Original comment by `oskar.wi...@gmail.com` on 18 Dec 2010 at 9:51

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Savegames for 0HP=dead & rest attached.

Auto-loading maps upon use would be good and during a game session they could 
be cached. A gaming session would begin when you launch the app and not with 
savegame load, but I assume it might be needed.
```

Original comment by `kims...@gmail.com` on 18 Dec 2010 at 10:48

**Comment by GoogleCodeExporter** [comment]
```
Forgot to attach :)
```

Original comment by `kims...@gmail.com` on 18 Dec 2010 at 10:48

Attachments:
- [savegame1](https://storage.googleapis.com/google-code-attachments/andors-trail/issue-87/comment-5/savegame1)
- [savegame2](https://storage.googleapis.com/google-code-attachments/andors-trail/issue-87/comment-5/savegame2)

**Comment by GoogleCodeExporter** [comment]
```
Thank you for the quick reply kimslot. 

I am now able to reproduce it, and have spotted the problem. This is a bug that 
will cause an ANR. I will try to find a solution for it as soon as possible.
```

Original comment by `oskar.wi...@gmail.com` on 18 Dec 2010 at 11:12
- Changed state: **Accepted**
- Added labels: **Milestone-0.6.7**

**Comment by GoogleCodeExporter** [comment]
```
A fix for this is now ready. It will make it into the next beta and the release 
version.

Thank you for reporting it!
```

Original comment by `oskar.wi...@gmail.com` on 18 Dec 2010 at 11:23
- Changed state: **Ready**

**Comment by GoogleCodeExporter** [comment]
```
That was fast :)
Too bad I cant play until then.

For anyone else, ANR means "Application Not Responding". I had to look it up :)
```

Original comment by `kims...@gmail.com` on 18 Dec 2010 at 11:38

**Comment by GoogleCodeExporter** [comment]
```
I also asked myself what ANR is. You may start a new character, right?
If I did get it right, then only your savegame is corrupted.

Not a very nice solution, indeed. ;-)
```

Original comment by `SamuelPl...@gmail.com` on 19 Dec 2010 at 12:00

**Comment by GoogleCodeExporter** [comment]
```
This is now fixed in v0.6.7b2, available from the download list.
```

Original comment by `oskar.wi...@gmail.com` on 19 Dec 2010 at 3:33

**Comment by GoogleCodeExporter** [comment]
Original comment by `oskar.wi...@gmail.com` on 20 Dec 2010 at 9:16
- Changed state: **Fixed**
